test/app_test.py

```diff
@@ -391,8 +392,49 @@ async def app_logs_pty(servicer, stream):
     with servicer.intercept() as ctx:
         ctx.set_responder("AppGetLogs", app_logs_pty)
 
-        with app.run(client=client):
-            time.sleep(0.1)
+        with OutputManager.enable_output():
+            with app.run(client=client):
+                time.sleep(0.1)
 
     captured = capsys.readouterr()
     assert captured.out.endswith("\nsome data\n\r")
+
+
+def test_show_progress_deprecations(client, monkeypatch):
```

Comment:
Would be more annoying but should we also be testing the actual logging / not logging behavior here too?

Reply:
I added some assertions that checks that the singleton is enabled!